.dotnet/src/Utility/StreamingClientResult.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+
+namespace OpenAI;
+
+/// <summary>
+/// Represents an operation response with streaming content that can be deserialized and enumerated while the response
+/// is still being received.
+/// </summary>
+/// <typeparam name="T"> The data type representative of distinct, streamable items. </typeparam>
+public class StreamingClientResult<T>
+    : IDisposable
```

Comment:
Should not implement IDisposable - see discussion in https://github.com/Azure/azure-sdk-for-net/pull/43392 and related email thread.